### ISSUE TITLE
Metaprogramming: `NullReferenceException` when exception is thrown in nested `AwaitForEach`

### ISSUE THREAD

**Issue by GerardSmit** [sakno/dotNext#46]
When an exception is thrown in `CodeGenerator.AwaitForEach` which contains another `CodeGenerator.AwaitForEach` (nested), you'll receive an `NullReferenceException` instead of the thrown exception.

Note: with a single `CodeGenerator.AwaitForEach` the exception that was being thrown in the body is being thrown, as expected.

## Information
.NET 5
DotNext.Metaprogramming 2.12.1

## Fiddle
Nested: https://dotnetfiddle.net/yfswYy (invalid, throws NullReferenceException)
Single: https://dotnetfiddle.net/DPnQrZ (valid, throws user exception)

## Example
```csharp
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using static DotNext.Metaprogramming.CodeGenerator;
					
public class Program
{
	public static async Task Main()
	{
		var result = AsyncLambda<Func<IAsyncEnumerable<char>, IAsyncEnumerable<char>, Task>>(fun => 
		{
			AwaitForEach(fun[0], ch =>
			{
				WriteLine(ch);
				AwaitForEach(fun[1], ch => WriteLine(ch));
			});
		}).Compile();
		
		await result(GetData('1'), GetData('2'));
	}
	
	public static async IAsyncEnumerable<char> GetData(char data)
	{
		await Task.Yield();
		yield return data;
		throw new Exception("Expected exception");
	}
}
```

### Expected
```
1
2
Unhandled exception. System.Exception: Expected exception
```

### Actual
```
1
2
Unhandled exception. System.NullReferenceException: Object reference not set to an instance of an object.
   at lambda_method2(Closure , AsyncStateMachine`1& )
   at DotNext.Runtime.CompilerServices.AsyncStateMachine`1.System.Runtime.CompilerServices.IAsyncStateMachine.MoveNext() in /_/src/DotNext.Metaprogramming/Runtime/CompilerServices/AsyncStateMachine.cs:line 124
--- End of stack trace from previous location ---
   at Program.Main()
   at Program.<Main>()
```

**Comment by sakno** [comment]
This is a bug in lowering algorithm when async expressions translated to regular LINQ expressions

**Comment by sakno** [comment]
@GerardSmit , could you please check the fix in develop branch?

**Comment by GerardSmit** [comment]
Is there a NuGet for CI builds or do I have to clone the repository locally?

Also; I'm impressed in how fast you found it and resolved it. Thanks a lot 😄

**Comment by sakno** [comment]
Clone is the best way.

**Comment by sakno** [comment]
Fixed in Metaprogramming 2.12.2 and published on NuGet.

**Comment by GerardSmit** [comment]
I've tested it with the published NuGet package and can confirm it has been fixed. Thanks!
